impeller/entity/contents/color_source_contents.h

```diff
@@ -118,6 +120,71 @@ class ColorSourceContents : public Contents {
                     typename VertexShaderT::FrameInfo frame_info,
                     const BindFragmentCallback& bind_fragment_callback) const {
     auto options = OptionsFromPassAndEntity(pass, entity);
+    options.primitive_type = geometry_result.type;
+
+    // Take the pre-populated vertex shader uniform struct and set managed
+    // values.
+    frame_info.depth = entity.GetShaderClipDepth();
+    frame_info.mvp = geometry_result.transform;
+
+    pass.SetVertexBuffer(std::move(geometry_result.vertex_buffer));
+
+    if constexpr (ContentContext::kEnableStencilThenCover) {
+      const bool is_stencil_then_cover =
+          geometry_result.mode == GeometryResult::Mode::kNonZero ||
+          geometry_result.mode == GeometryResult::Mode::kEvenOdd;
+      if (is_stencil_then_cover) {
+        pass.SetStencilReference(0);
+
+        /// Stencil preparation draw.
```

Comment:
What is the stencil preparation?

Reply:
Oh i c, I think I'm just having a hard time following this code now. This is essentially taking whatever geometry we got and configuring how to treat it, in this case it goes to the stencil buffer. Then we add the cover + reset afterwards. makes sense!